cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -0,0 +1,462 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import namedtuple
+
+from aws_cdk import aws_cloudwatch as cloudwatch
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import core
+
+from pcluster.models.cluster_config import BaseClusterConfig, SharedStorageType
+
+
+class Coord:
+    """Create coordinates for locating cloudwatch graphs."""
+
+    def __init__(self, x_value: int, y_value: int):
+        self.x_value = x_value
+        self.y_value = y_value
+
+
+GeneralMetric = namedtuple("GeneralMetric", ["title", "metrics", "supported_vol_types"], defaults=[None, None])
```

Comment:
What about PclusterMetric?

Reply:
Then we need to remove defaults, because it's not supported by pcluster 3.6 and we need to support it.
You can use an approach like:
```python
_PclusterMetric = namedtuple(...)

def new_metric(title=None, metrics=None, supported_vol_types=None):
    return _PclusterMetric(val, left, right)
```
and use new_metric everywhere.
The alternative is to create a class for PclusterMetric too.

Reply:
Done